prow/github/client_test.go

```diff
@@ -2980,45 +2983,83 @@ func TestAllMethodsThatDoRequestSetOrgHeader(t *testing.T) {
 				}
 				return &http.Response{Body: ioutil.NopCloser(&bytes.Buffer{})}, nil
 			}}
-
 			ghClient.(*client).client.(*http.Client).Transport = checkingRoundTripper
 			ghClient.(*client).gqlc.(*graphQLGitHubAppsAuthClientWrapper).Client = githubv4.NewClient(&http.Client{Transport: checkingRoundTripper})
-			clientValue := reflect.ValueOf(ghClient)
-
-			var args []reflect.Value
-			// First arg is self, so start with second arg
-			for j := 1; j < clientType.Method(i).Func.Type().NumIn(); j++ {
-				arg := reflect.New(clientType.Method(i).Func.Type().In(j)).Elem()
-				if arg.Kind() == reflect.Ptr && arg.IsNil() {
-					arg.Set(reflect.New(arg.Type().Elem()))
-				}
 
-				if arg.Type() == stringType {
-					arg.Set(stringValue)
-				}
+			// We don't care about the result at all, the verification happens via the roundTripper
+			_ = call()
+		})
+	}
+}
 
-				// We can not deal with interface types genererically, as there
-				// is no automatic way to figure out the concrete values they
-				// can or should be set to.
-				if arg.Type().String() == "context.Context" {
-					arg.Set(reflect.ValueOf(context.Background()))
-				}
-				if arg.Type().String() == "interface {}" {
-					arg.Set(reflect.ValueOf(map[string]interface{}{}))
-				}
+func getCallForAllClientMethodsThroughReflection(
```

Comment:
I factored this out from `TestAllMethodsThatDoRequestSetOrgHeader` so that I can test the context handling through reflect by checking on all methods whose name ends with `WithContext` rather than having to write a test by hand for each of them.